docs/DeveloperGuide.md

```diff
@@ -1,31 +1,400 @@
+
+**Finding a topic in TopicList:**
+
+This task is performed by the `TopicList.find()` method. 
+
+Step 1: The `parseCommand()` method instantiates a `TopicsCommand` object which then calls the `TopicList.find()` 
+        method. A String object derived from the user's input is passed into this method.
+
+Step 2: The `TopicList.find()` method checks if any `Topic` object in the `TopicList` has a description that contains 
+        the `String` object passed into the method. Such `Topic` objects are stored in a new `TopicList`.
+
+Step 3: The `TopicList.find()` method then calls the `TopicList.showTopics()` method, passing in the new TopicList. The 
+        `Ui.printTopicList()` method is called within the `TopicList.showTopics()` method, printing out topic 
+        descriptions containing the user's input.
+
+`NoteList` is also an `ArrayList` of type `Displayable`, which is one of two interfaces implemented in the code for 
+E-Duke-8. As such, many of the commands that manipulate the `TopicList` make use of the package `java.util.ArrayList`. 
+The `NoteList` stores `Note` objects. Each topic has 1 `NoteList`. 
+
+#### 2.2.3. Implementation of Notes
+
+**Adding a new note:**
+
+This task is performed by the `NoteList.add()` method.
+
+Step 1: The `parseCommand()` method instantiates a `NoteCommand` object which then calls the `NoteList.add()` method. 
+        A new `Note` object is passed into its parameter.
+
+Step 2: The `NoteList.add()` method makes use of `ArrayList` API, specifically the `ArrayList.add()` method, to add 
+        the `Note` object into `NoteList`.
+
+**Deleting a note:**
+
+This task is performed by the `NoteList.add()` method.
+
+Step 1: The `parseCommand()` method instantiates a `NoteCommand` object which then calls the `NoteList.delete()` method. 
+        An integer that represents the index of the `Note` object to be deleted within the `NoteList` is passed into 
+        this method.
+
+Step 2: The `NoteList.add()` method makes use of `ArrayList` API, specifically the `ArrayList.remove()` method, to 
+        delete the `Note` object in `NoteList`.
+
+**Listing out all notes in a topic**
+
+This task is performed by the `Topic.showNotes()` method.
+
+Step 1: The `parseCommand()` method instantiates a `TopicCommand` object which then calls the `Topic.showNotes()` method. 
+
+Step 2: The `Topic.showNotes()` method calls the `Ui.printNoteList()` method. The topic's `NoteList` into this method. 
+        `Ui.printNoteList()` prints out all the descriptions of the `Note` objects in the `NoteList`.
+
+#### 2.2.4. Design of Option and OptionList 
+
+The `Option` and `OptionList` classes implements the `Displayable` and `DisplayableList` interfaces respectively. 
+The `Option` object stores one option of a question while the `OptionList` object stores all 4 options of the same 
+question. The class diagram below illustrates the structure of both classes. 
+
+![Option_and_OptionList_Class](./images/Option.png)
```

Comment:
![Screenshot 2020-10-28 at 12 21 52 PM](https://user-images.githubusercontent.com/42382015/97390616-40ae3400-1918-11eb-86f2-d303e9e47ab4.jpg)

Since OptionList is holding Option, you might have missed out on some link or dependency between the 2.

Reply:
Sorry for the confusion, but in our implementation options implements displayable and OptionList is actually holding onto displayable now.